src/components/views/DownloadModModal.vue

```diff
@@ -15,56 +15,14 @@
-                    Select a version of {{thunderstoreMod.getName()}} to download
-                </h2>
-            </template>
-            <template v-slot:body>
-                <p>It's recommended to select the latest version of all mods.</p>
-                <p>Using outdated versions may cause problems.</p>
-                <br/>
-                <div class="columns is-vcentered">
-                    <template v-if="currentVersion !== null">
-                        <div class="column is-narrow">
-                            <select class="select" disabled="true">
-                                <option selected>
-                                    {{currentVersion}}
-                                </option>
-                            </select>
-                        </div>
-                        <div class="column is-narrow">
-                            <span class="margin-right margin-right--half-width"><span class="margin-right margin-right--half-width"/> <i class='fas fa-long-arrow-alt-right'></i></span>
-                        </div>
-                    </template>
-                    <div class="column is-narrow">
-                        <select class='select' v-model='selectedVersion'>
-                            <option v-for='(value, index) in versionNumbers' :key='index' v-bind:value='value'>
-                                {{value}}
-                            </option>
-                        </select>
-                    </div>
-                    <div class="column is-narrow">
-                        <span class="tag is-dark" v-if='selectedVersion === null'>
-                            You need to select a version
-                        </span>
-                        <span class="tag is-success" v-else-if='recommendedVersion === selectedVersion'>
-                            {{selectedVersion}} is the recommended version
-                        </span>
-                        <span class="tag is-success" v-else-if='versionNumbers[0] === selectedVersion'>
-                            {{selectedVersion}} is the latest version
-                        </span>
-                        <span class="tag is-danger" v-else-if='versionNumbers[0] !== selectedVersion'>
-                            {{selectedVersion}} is an outdated version
-                        </span>
-                    </div>
-                </div>
-            </template>
-            <template v-slot:footer>
-                <button class="button is-info" @click="downloadThunderstoreMod()">Download with dependencies</button>
-            </template>
-        </ModalCard>
+        <DownloadModVersionSelectModal
+            :versionNumbers="versionNumbers"
+            :selected-version="selectedVersion"
+            :currentVersion="currentVersion"
+            :recommendedVersion="recommendedVersion"
+            :download-thunderstore-mod="downloadThunderstoreMod"
```

Comment:
Now, if you followed the idea from the earlier comment, and it worked, it's quite easy to see that the only reason the parent component is interested in `selectedVersion` is that it's needed by `downloadThunderstoreMod`. Furthermore, the parent component doesn't care about `downloadThunderstoreMod` as it's only passed to the child component, which means that it should reside in the child component, not the parent.

If that also works without problems, the next step in chain would be `downloadHandler`, which is actually also only used by child component. However, the internal implementation of `downloadHandler` is tightly coupled with the parent component, so it might be a good spot limit the scope of this PR and leave it into the parent.

That then leaves the question of how `DownloadModVersionSelectModal.downloadThunderstoreMod` should trigger `DownloadModModal.downloadHandler`? For this PR I'd say the subcomponent should emit an event with a `ThunderstoreCombo` as a payload, and the parent should catch the event and call `downloadHandler`. If you're thinking that Vuex could be used for that as well, you're right, but lets leave that for another PR.

Reply:
yeah, I think the downloadHandler should be _handled_ 🥁 by vuex once it is time for that.